use-timescale/continuous-aggregates/about-continuous-aggregates.md

```diff
@@ -17,6 +17,88 @@ import CaggsTypes from "versionContent/_partials/_caggs-types.mdx";
+    the materialization hypertable
+*   Invalidation engine to determine when data needs to be re-materialized, due
+    to changes in the data
+*   Query engine to access the aggregated data
+
+### Materialization hypertable
+
+Continuous aggregates take raw data from the original hypertable, aggregate it,
+and store the intermediate state in a materialization hypertable. When you query
+the continuous aggregate view, the state is returned to you as needed.
+
+Using the same temperature example, the materialization table looks like this:
+
+|day|location|chunk|avg temperature partial|
+|-|-|-|-|
+|2021/01/01|New York|1|{3, 219}|
+|2021/01/01|Stockholm|1|{4, 280}|
+|2021/01/02|New York|2||
+|2021/01/02|Stockholm|2|{5, 345}|
+
+The materialization table is stored as a Timescale hypertable, to take
+advantage of the scaling and query optimizations that hypertables offer.
+Materialization tables contain a column for each group-by clause in the query,
+a `chunk` column identifying which chunk in the raw data this entry came from,
+and a `partial aggregate` column for each aggregate in the query.
+
+The partial column is used internally to calculate the output. In this example,
+because the query looks for an average, the partial column contains the number
+of rows seen, and the sum of all their values. The most important thing to know
+about partials is that they can be combined to create new partials spanning all
+of the old partials' rows. This is important if you combine groups that span
+multiple chunks.
+
+For more information, see [materialization hypertables][cagg-mat-hypertables].
+
+### Materialization engine
+
+The materialization engine performs two transactions. The first transaction
+blocks all INSERTs, UPDATEs, and DELETEs, determines the time range to
+materialize, and updates the invalidation threshold. The second transaction
+unblocks other transactions, and materializes the aggregates. The first
+transaction is very quick, and most of the work happens during the second
+transaction, to ensure that the work does not interfere with other operations.
+
+When you query the continuous aggregate view, the materialization engine
+combines the aggregate partials into a single partial for each time range, and
+calculates the value that is returned. For example, to compute an average, each
+partial sum is added up to a total sum, and each partial count is added up to a
+total count, then the average is computed as the total sum divided by the total
+count.
+
+### Invalidation engine
+
+Any change to the data in a hypertable could potentially invalidate some
+materialized rows. The invalidation engine checks to ensure that the system does
+not become swamped with invalidations.
+
+Fortunately, time-series data means that nearly all INSERTs and UPDATEs have a
+recent timestamp, so the invalidation engine does not materialize all the data,
+but to a set point in time called the materialization threshold. This threshold
```

Comment:
The internal term for a threshold is "watermark", maybe we want to put this into parenthesis.